include/RAJA/policy/sycl/forall.hpp

```diff
@@ -210,11 +218,85 @@ RAJA_INLINE resources::EventProxy<resources::Sycl> forall_impl(resources::Sycl &
     RAJA_FT_END;
   }
 
-  
-
   return resources::EventProxy<resources::Sycl>(sycl_res);
 }
 
+template <typename Iterable, typename LoopBody, size_t BlockSize, bool Async, typename ForallParam,
+          typename std::enable_if<std::is_trivially_copyable<LoopBody>{},bool>::type = true>
+RAJA_INLINE
+concepts::enable_if_t< 
+  resources::EventProxy<resources::Sycl>,
+  RAJA::expt::type_traits::is_ForallParamPack<ForallParam>,
+  concepts::negate<RAJA::expt::type_traits::is_ForallParamPack_empty<ForallParam>> >
+forall_impl(resources::Sycl &sycl_res,
```

Comment:
what about the case where the LoopBody is not trivially copyable and f_params is not empty?

Reply:
I've added the case where LoopBody is not trivially copyable and f_params is not empty.